OpenRA.Mods.Common/Widgets/Logic/Settings/DisplaySettingsLogic.cs

```diff
@@ -269,16 +269,14 @@ Func<bool> InitPanel(Widget panel)
 			};
 
 			var colorManager = modData.DefaultRules.Actors[SystemActors.World].TraitInfo<ColorPickerManagerInfo>();
-			colorManager.Color = ps.Color;
 
-			var colorDropdown = panel.Get<DropDownButtonWidget>("PLAYERCOLOR");
-			colorDropdown.IsDisabled = () => worldRenderer.World.Type != WorldType.Shellmap;
-			colorDropdown.OnMouseDown = _ => ColorPickerLogic.ShowColorDropDown(colorDropdown, colorManager, worldRenderer, () =>
-			{
-				Game.Settings.Player.Color = colorManager.Color;
-				Game.Settings.Save();
-			});
-			colorDropdown.Get<ColorBlockWidget>("COLORBLOCK").GetColor = () => ps.Color;
+			SettingsUtils.BindColorWidget(panel, worldRenderer, colorManager, () => worldRenderer.World.Type != WorldType.Shellmap, "PLAYERCOLOR", Game.Settings.Player, "Color");
```

Comment:
Isn't it a bit odd that one is in `Game.Settings.Player` and the others are in `Game.Settings.Game`?

Reply:
No, it's the main color of the player, nothing to do with these new features. I just made it's code reusable